modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/rel/ModifyNode.java

```diff
@@ -268,15 +268,22 @@ private void inlineUpdates(int offset, List<RowT> rows) {
 
         RowHandler<RowT> handler = context().rowHandler();
 
-        for (RowT row : rows) {
-            for (int i = 0; i < mapping.length; i++) {
-                if (offset == 0 && i == mapping[i]) {
-                    continue;
-                }
+        rows.replaceAll(row -> handler.map(row, applyOffset(mapping, offset)));
```

Comment:
it's better to do computation outside of the loop

Reply:
Done, thanks.